FRBDK/Glue/OfficialPlugins/AnimationChainPlugin/Views/AchxPreviewView.xaml.cs

```diff
@@ -196,6 +211,183 @@ void CreatePolygonsFor(AnimationChainSave animation)
             }
         }
 
+        private void RefreshAnimation()
+        {
+            var texture = MainAnimationSprite.Texture;
+            
+            _currentAnimationChain = null;
+            _currentAnimationFrame = -1;
+            _lastFrameTime = DateTime.MinValue;
+            _currentFrameTime = 0;
+
+            foreach(var shape in AnimationShapes)
+            {
+                GumCanvasAnimation.Children.Remove(shape);
+            }
+            AnimationShapes.Clear();
+
+            if (texture != null && ViewModel != null)
+            {
+                if (ViewModel.SelectedAnimationFrame != null || ViewModel.SelectedShape != null)
+                {
+                    var frame = ViewModel.SelectedAnimationFrame ?? ViewModel.SelectedShape.Parent;
+                    MainAnimationSprite.Width = 100;
+                    MainAnimationSprite.Height = 100;
+                    MainAnimationSprite.TextureAddress = Gum.Managers.TextureAddress.Custom;
+                    MainAnimationSprite.TextureLeft = (int)frame.LeftCoordinate;
+                    MainAnimationSprite.TextureTop = (int)frame.TopCoordinate;
+                    MainAnimationSprite.TextureWidth = FlatRedBall.Math.MathFunctions.RoundToInt(frame.RightCoordinate - frame.LeftCoordinate);
+                    MainAnimationSprite.TextureHeight = FlatRedBall.Math.MathFunctions.RoundToInt(frame.BottomCoordinate - frame.TopCoordinate);
+                    MainAnimationSprite.WidthUnits = Gum.DataTypes.DimensionUnitType.PercentageOfSourceFile;
+                    MainAnimationSprite.HeightUnits = Gum.DataTypes.DimensionUnitType.PercentageOfSourceFile;
+
+                    MainAnimationSprite.Visible = true;
+
+                    List<ShapeViewModel> shapes;
+                    if(ViewModel.SelectedShape != null)
+                    {
+                        shapes = new List<ShapeViewModel>()
+                        {
+                            ViewModel.SelectedShape
+                        };
+                    }
+                    else
+                    {
+                        shapes = ViewModel.SelectedAnimationFrame.VisibleChildren.ToList();
+                    }
+
+                    renderShapes(shapes);
+
+                    FocusAnimation();
+                }
+                else if (ViewModel.SelectedAnimationChain != null)
+                {
+                    if (ViewModel.SelectedAnimationChain.VisibleChildren.Count > 0)
+                    {
+                        _currentAnimationChain = ViewModel.SelectedAnimationChain;
+
+                        RunAnimation();
+                    }
+                }
+                else //if(ViewModel.SelectedAnimationChain == null)
+                {
+                    MainAnimationSprite.Visible = false;
+                }
+            }
+        }
+
+        private void renderShapes(List<ShapeViewModel> shapes)
```

Comment:
Capitalize method? Even though it's private.

Reply:
Haha.  That's javascript syntax.  Sometimes it's hard to keep your syntax conventions straight.